internal/controller/dnsrecord_controller.go

```diff
@@ -387,19 +408,22 @@ func (r *DNSRecordReconciler) applyChanges(ctx context.Context, dnsRecord *v1alp
 				return noRequeueDuration, err
 			}
 		}
+
 		dnsRecord.Status.ValidFor = validationRequeueTime.String()
+		dnsRecord.Status.QueuedAt = reconcileStart
 		setDNSRecordCondition(dnsRecord, string(v1alpha1.ConditionTypeReady), metav1.ConditionFalse, "AwaitingValidation", "Awaiting validation")
-		logger.Info("Applying changes")
-		err = registry.ApplyChanges(ctx, plan.Changes)
-		if err != nil {
-			return validationRequeueTime, err
-		}
-	} else {
-		logger.Info("All records are already up to date")
-		dnsRecord.Status.WriteCounter = 0
-		wrtiteCounter.WithLabelValues(dnsRecord.Name, dnsRecord.Namespace).Set(0)
-		setDNSRecordCondition(dnsRecord, string(v1alpha1.ConditionTypeReady), metav1.ConditionTrue, "ProviderSuccess", "Provider ensured the dns record")
+
+		logger.Info(fmt.Sprintf("endpoints were updated, requing for validation in %s", validationRequeueTime.String()))
+		return validationRequeueTime, nil
 	}
 
+	//Set WriteCounter, ValidFor, QueuedAt, Ready condition true and return defaultRequeueTime when we have no changes!
+	dnsRecord.Status.WriteCounter = 0
```

Comment:
WriteCounter shouldn't be getting reset here either, right? It should only reset to 0 when the generation changes

Reply:
This is the same logic as main